packages/datetime/src/dateFormatter.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ */
+
+/**
+ * Allows arbitrary formatting of dates in the @blueprintjs/datetime inputs.
+ */
+export interface IDateFormatter {
+    /**
+     * Optional placeholder to show in inputs whose content will be passed to this formatter.
```

Comment:
This language feels a little clunky to me.

```tsx
/**
 * Placeholder text to display in empty input fields.
 * Recommended practice is to indicate the date format that `stringToDate` 
 * expects.
 */
```

Second sentence is optional but maybe helpful.